exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFileSystem.java

```diff
@@ -89,22 +89,36 @@ public DrillFileSystem(Configuration fsConf) throws IOException {
   }
 
   public DrillFileSystem(Configuration fsConf, OperatorStats operatorStats) throws IOException {
-    this.underlyingFs = FileSystem.get(fsConf);
+    this(fsConf, URI.create(fsConf.getRaw(FS_DEFAULT_NAME_KEY)), operatorStats);
+  }
+
+  public DrillFileSystem(Configuration fsConf, URI Uri, OperatorStats operatorStats) throws IOException {
+    this.underlyingFs = FileSystem.get(Uri, fsConf);
+    logger.trace("Configuration for the DrillFileSystem " + fsConf.getRaw(FS_DEFAULT_NAME_KEY) +
+        ", underlyingFs: " + this.underlyingFs.getUri());
     this.codecFactory = new CompressionCodecFactory(fsConf);
     this.operatorStats = operatorStats;
+    setConf(fsConf);
```

Comment:
I think this whole code block needs more thought. DrillFileSystem derives from the Hadoop `FileSystem` class, but we never invoke the `FileSystem` constructor. Why? Without proper init, can we guarantee that the methods defined in the base class actually work?

On the other hand, it seems we've reimplemented all base class methods, basically masking the base class implementation. If that is true, why set the config on the base class?

Reply:
Agreed I will remove this call, and make setConf effectively a noop which will throw an exception if anyone calls it.